### ISSUE TITLE
Contour can spill into null area

### ISSUE THREAD

**Issue by etpinard** [plotly/plotly.js#4200]
See https://community.plot.ly/t/contour-plot-regions-bleeding-into-null-area/28399 and https://codepen.io/ucode/pen/ExYRBZX

where

```js
  {
    type: 'contour',
    line: { smoothing: 0 },
     x: [0, 1],
     y: [0, 1],
     z:
        [ [ 0, 1 ]
        , [ null, 0 ]
        ]
  }
```

gives

![image](https://user-images.githubusercontent.com/6675409/65152584-3ff95d80-d9f6-11e9-8871-a25bc3a8e569.png)

Here, the darkest blue filled contour should (probably) not spill above the `y=x` line.

**Comment by alexcjohnson** [comment]
This is a consequence of the small width we give around valid points, so they're still visible if completely surrounded by nulls. See for example https://rreusser.github.io/plotly-mock-viewer/#connectgaps_2d

<img width="361" alt="Screen Shot 2019-09-18 at 9 47 15 AM" src="https://user-images.githubusercontent.com/2678795/65154463-9025ef00-d9f9-11e9-9a4d-d98d0b443dee.png">

Without this extra width, the lines at the top, bottom, and right would disappear entirely.

**Comment by samueller** [comment]
@alexcjohnson is it possible to programmatically disable the small extra width around valid points?

**Comment by alexcjohnson** [comment]
We could certainly add such an option. It's currently hardcoded here:

https://github.com/plotly/plotly.js/blob/7ec0729b1dd0ff10b6056197d6ddf8552d1f7862/src/traces/contour/plot.js#L632-L636

**Comment by samueller** [comment]
@alexcjohnson that solved it! I simply changed this to `level: 0.999` and everything looks beautiful. Setting `level` to `1` made everything disappear, I'm guessing because I only have points in the corners.

**Comment by gvwilson** [comment]
Hi - we are currently trying to tidy up Plotly's public repositories to help us focus our efforts on things that will help users most. Since this issue has been sitting for several years, I'm going to close it. If it's still a concern, we'd be grateful if you could open a new issue (with a short reproducible example if appropriate) so that we can add it to our backlog. Thanks for your help - @gvwilson
